log4j2-elasticsearch-core/src/main/java/org/appenders/log4j2/elasticsearch/VirtualPropertiesWriter.java

```diff
@@ -112,7 +114,7 @@ public void serializeAsField(Object bean, JsonGenerator gen, SerializerProvider
 
             VirtualProperty property = virtualProperties[i];
 
-            String resolved = valueResolver.resolve(property);
+            String resolved = valueResolver.resolve(property, (bean instanceof LogEvent) ? (LogEvent) bean : null);
```

Comment:
Unsure about this: will `bean` ever not be a `LogEvent`? In that case it's probably better to cast and get an exception?